src/GitVersion.LibGit2Sharp/Git/ObjectId.cs

```diff
@@ -2,26 +2,24 @@
 
 namespace GitVersion
 {
-    internal class ObjectId : IObjectId
+    internal sealed class ObjectId : IObjectId
     {
-        private static readonly LambdaEqualityHelper<IObjectId> equalityHelper =
-            new LambdaEqualityHelper<IObjectId>(x => x.Sha);
+        private static readonly LambdaEqualityHelper<IObjectId> equalityHelper = new(x => x.Sha);
+        private static readonly LambdaKeyComparer<IObjectId, string> comparerHelper = new(x => x.Sha);
 
         private readonly LibGit2Sharp.ObjectId innerObjectId;
-        internal ObjectId(LibGit2Sharp.ObjectId objectId)
-        {
-            innerObjectId = objectId;
-        }
+        internal ObjectId(LibGit2Sharp.ObjectId objectId) => innerObjectId = objectId;
 
         public ObjectId(string sha) : this(new LibGit2Sharp.ObjectId(sha))
         {
         }
 
-        public override bool Equals(object obj) => Equals(obj as IObjectId);
+        public int CompareTo(IObjectId other) => comparerHelper.Compare(this, other);
+        public override bool Equals(object obj) => Equals((obj as IObjectId)!);
         public bool Equals(IObjectId other) => equalityHelper.Equals(this, other);
         public override int GetHashCode() => equalityHelper.GetHashCode(this);
-        public static implicit operator LibGit2Sharp.ObjectId(ObjectId d) => d?.innerObjectId;
-        public string Sha => innerObjectId?.Sha;
+        public static implicit operator LibGit2Sharp.ObjectId(ObjectId d) => d.innerObjectId;
+        public string Sha => innerObjectId.Sha;
 
         public string ToString(int prefixLength) => innerObjectId.ToString(prefixLength);
```

Comment:
Should we provide an `override ToString()` method that does not take `prefixLength` as well?

Reply:
Will add

Reply:
Done